icubam/backoffice/handlers/users.py

```diff
@@ -39,50 +40,111 @@ def get(self):
       '{}?id={}'.format(UserHandler.ROUTE, self.user.user_id))
 
 
-
 class UserHandler(base.BaseHandler):
   ROUTE = "/user"
 
+  """Creates ICU DTOs based on ALL ICUs and those from the subset."""
+  def prepare_icus(self, subset_icus_ids):
+    icus = self.store.get_icus()
+    output = []
+    for icu in icus:
+      dto = icu.to_dict()
+      dto["selected"] = icu.icu_id in subset_icus_ids
+      output.append(dto)
+    return sorted(output, key = lambda x: x['name'])
+
+  """Prepares the DTO comparing ALL icus against the ones from the user."""
+  def prepare_icus_for_user(self, user):
+    return self.prepare_icus([user_icu.icu_id for user_icu in user.icus])
+
   @tornado.web.authenticated
   def get(self):
     userid = self.get_query_argument('id', None)
     user = None
     if userid is not None:
-      user = self.db.get_user(userid)
-    # TODO(fred): load the ICUs to show in the form.
+      user = self.store.get_user(userid)
+
     user = user if user is not None else store.User()
-    self.render("user.html", icus=[], user=user, error="")
+    icus_dto = self.prepare_icus_for_user(user)
+    self.render(
+      "user.html", icus=icus_dto, user=user, error=False,
+      allow_admin=self.user.is_admin)
```

Comment:
you can remove `allow_admin`, and in the form simply user {{this_user.is_admin}}. I have overloaded self.render to always inject this_user=self.user :)

Reply:
hahahaha I see it now in BaseHandler.
G'enial.